tests/unit/test_routes/test_resource_retreival.py

```diff
@@ -0,0 +1,186 @@
+
+
+def test_get_favicon(module_client):
+    response = module_client.get("favicon.ico")
+    assert (response.status_code == 200)
+
+
+def test_get_docs(module_client):
+    response = module_client.get("/")
+    assert (response.status_code == 200)
+
+
+def test_open_api_yaml(module_client):
+    response = module_client.get("/openapi.yaml")
+    assert (response.status_code == 200)
+    open_api_yaml = load(response.data)
+    assert (isinstance(open_api_yaml, dict))
+    assert (open_api_yaml.get("info").get("version") == LATEST_API_VERSION)
+
+
+def test_paid_filter(module_client, module_db):
+    client = module_client
+
+    total_resources = client.get('api/v1/resources').json['total_count']
+
+    # Filter by paid
+    response = client.get('api/v1/resources?paid=false')
+
+    total_free_resources = response.json['total_count']
+
+    assert all([not res.get('paid') for res in response.json['data']])
+
+    response = client.get('api/v1/resources?paid=true')
+
+    total_paid_resources = response.json['total_count']
+
+    assert all([res.get('paid') for res in response.json['data']])
+
+    # Check that the number of resources appear correct
+    assert (total_paid_resources > 0)
+    assert (total_free_resources > 0)
+    assert (total_resources == total_free_resources + total_paid_resources)
+
+
+def test_paid_filter_works_with_uppercase_parameter(module_client, module_db):
+    client = module_client
+
+    response = client.get('api/v1/resources?paid=TRUE')
+    assert all([res.get('paid') for res in response.json['data']])
+
+    response = client.get('api/v1/resources?paid=FALSE')
+    assert all([not res.get('paid') for res in response.json['data']])
+
+
+def test_paid_filter_defaults_all_when_invalid_paid_parameter(module_client, module_db):
+    client = module_client
+
+    response = client.get('api/v1/resources?paid=na93ns8i1ns')
+
+    assert (True in [res.get('paid') for res in response.json['data']])
+    assert (False in [res.get('paid') for res in response.json['data']])
+
+
+def test_filters(module_client, module_db):
+    client = module_client
+
+    # Filter by one language
+    response = client.get('api/v1/resources?languages=python')
+
+    for resource in response.json['data']:
+        assert (isinstance(resource.get('languages'), list))
+        assert ('Python' in resource.get('languages'))
+
+    # Filter by multiple languages
+    response = client.get('api/v1/resources?languages=python&languages=javascript')
+
+    for resource in response.json['data']:
+        assert (isinstance(resource.get('languages'), list))
+        assert (
+            ('Python' in resource.get('languages')) or
+            ('JavaScript' in resource.get('languages'))
+        )
+
+    # Filter by category
+    response = client.get('api/v1/resources?category=Back%20End%20Dev')
+
+    for resource in response.json['data']:
+        assert (resource.get('category') == "Back End Dev")
+
+    # TODO: Filter by updated_after
```

Comment:
Maybe we should go ahead and spend the time to try to figure this one out as well. Or just cut our losses and delete this TODO without that extra test